### PR TITLE
Fixes and tests for julia-0.5

### PR DESCRIPTION
**Pull request by timholy**
As predicted in #4, this needed another update. This time, I suspect this should "stick" a little longer.

I also added a "real" test. The patch file is at greatest risk for quickly going out-of-date, but it still seems like a step forward.

### CODE REVIEW

**Comment by timholy** [comment]
Looks like Travis needs to be turned on. If you do that, I can do a force push to get it to run.

File: test/inference_hacked.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
++# precompile(typeof(typeinf_edge).name.mt.defs.sig)
+ 
+ for m in _methods_by_ftype(Tuple{typeof(typeinf_loop), Vararg{Any}}, 10)
+     typeinf(m[3], m[1], m[2], true)
+@@ -3423,4 +3423,4 @@
+     typeinf(m[3], m[1], m[2], true)
+ end
+ 
+-ccall(:jl_set_typeinf_func, Void, (Any,), typeinf_ext)
++# ccall(:jl_set_typeinf_func, Void, (Any,), typeinf_ext)
```
Comment (vtjnash) [inline test/inference_hacked.patch]: this line can move from inference.jl to coreimg.jl in base

File: test/inference_hacked.patch [hunk 2]
```diff
@@ -0,0 +1,17 @@
+--- inference.jl	2016-05-07 11:15:48.310879925 -0500
++++ inference_hacked.jl	2016-05-07 15:16:28.905769431 -0500
+@@ -3414,7 +3414,7 @@
+ # make sure that typeinf is executed before turning on typeinf_ext
+ # this ensures that typeinf_ext doesn't recurse before it can add the item to the workq
+ 
+-precompile(typeof(typeinf_edge).name.mt.defs.sig)
++# precompile(typeof(typeinf_edge).name.mt.defs.sig)
```
Comment (vtjnash) [inline test/inference_hacked.patch]: this line is a no-op when run on master, so we can just delete it

**Comment by ihnorton** [comment]
> Looks like Travis needs to be turned on.

Done.

You have push access -- since you are actually using this right now, please feel free to merge, push without PR, go wild, etc.

(If there's a better place for either code or repo I'm fine with moving it)

**Comment by timholy** [comment]
I think it's best leaving it here, as long as you don't mind doing occasional administrative stuff. I'm happy to merge my own patches, too, just so it doesn't steal your bandwidth.

This may not need many more changes other than tracking base. The only thing I can think of now is to come up with some scheme to preserve the line numbers in the source code.

**Comment by timholy** [comment]
With https://github.com/JuliaLang/julia/pull/16256, the patch is no longer necessary. Thanks @vtjnash, that was a much nicer solution.

**Pull request merged**